packages/cysync-core/src/context/buySell/index.tsx

```diff
@@ -194,44 +195,54 @@ export const BuySellProvider: FC<BuySellContextProviderProps> = ({
     setUnhandledError(e);
   }, []);
 
+  const onRetry = useCallback(() => {
+    setState(BuySellState.CURRENCY_SELECT);
+    resetUserInput();
+  }, []);
+
   const initHandler = useCallback(async () => {
-    const pairs = await buySellSupport.getTradingPairs();
-    tradingPairs.current = pairs;
-
-    const fiatDropdown: DropDownItemProps[] = pairs.fiatCurrencies.map(f => ({
-      id: f.code,
-      checkType: 'radio',
-      leftImage: (
-        <Typography $fontSize={24}>{f.currency.countryFlag}</Typography>
-      ),
-      text: f.currency.name,
-      shortForm: `(${f.currency.code.toUpperCase()})`,
-    }));
-    setFiatDropdownList(fiatDropdown);
-
-    const cryptoDropdown: DropDownItemProps[] = pairs.cryptoCurrencies.map(
-      c => {
-        const parentId = (c.coin as IEvmErc20Token)?.parentId;
-        return {
-          id: c.coin.id,
-          checkType: 'radio',
-          leftImage: (
-            <CoinIcon
-              assetId={c.coin.id}
-              parentAssetId={(c.coin as IEvmErc20Token).parentId ?? c.coin.id}
-            />
-          ),
-          text: c.coin.name,
-          shortForm: `(${c.coin.abbr})`,
-          rightText: parentId
-            ? parentId[0].toUpperCase() + parentId.slice(1).toLowerCase()
-            : '',
-        };
-      },
-    );
+    try {
+      const pairs = await buySellSupport.getTradingPairs();
+      tradingPairs.current = pairs;
+
+      const fiatDropdown: DropDownItemProps[] = pairs.fiatCurrencies.map(f => ({
+        id: f.code,
+        checkType: 'radio',
+        leftImage: (
+          <Typography $fontSize={24}>{f.currency.countryFlag}</Typography>
+        ),
+        text: f.currency.name,
+        shortForm: `(${f.currency.code.toUpperCase()})`,
+      }));
+      setFiatDropdownList(fiatDropdown);
+
+      const cryptoDropdown: DropDownItemProps[] = pairs.cryptoCurrencies.map(
+        c => {
+          const { parentId } = c.coin as IEvmErc20Token;
+          return {
+            id: c.coin.id,
+            checkType: 'radio',
+            leftImage: (
+              <CoinIcon
+                assetId={c.coin.id}
+                parentAssetId={(c.coin as IEvmErc20Token).parentId ?? c.coin.id}
+              />
+            ),
+            text: c.coin.name,
+            shortForm: `(${c.coin.abbr})`,
+            rightText: parentId
+              ? parentId[0].toUpperCase() + parentId.slice(1).toLowerCase()
+              : '',
+          };
+        },
+      );
 
-    setCryptoDropdownList(cryptoDropdown);
-    return true;
+      setCryptoDropdownList(cryptoDropdown);
+      return true;
+    } catch (err) {
+      onError(err);
+      return false;
```

Comment:
All errors are piped directly to `unhandledError`, have you considered what will happen if the error is not a server error?

Reply:
Will pick in a different PR